### ISSUE TITLE
x/tools/gopls: embed errors don't always appear

### ISSUE THREAD

**Issue by heschi** [golang/go#44342]
This program shows no error on the embed pattern:

```
package embedtest

import (
	_ "embed"
)

//go:embed NONEXISTANT
var foo string
```

When I add an import, the error appears, but only if it's not in the stdlib:

```
package embedtest

import (
	_ "embed"

	_ "golang.org/x/tools/imports"
)

//go:embed NONEXISTANT
var foo string
```

At first I thought this was because we should invalidate metadata when a `//go:embed` comment changes, but even when I restart gopls the IWL doesn't pick up the error. So there's something weird going on here that I didn't dig into. (We should probably still invalidate metadata on embed comment changes, or implement embed pattern checking ourselves.)

Reported by @myitcv in Slack.

**Comment by heschik** [comment]
https://github.com/golang/go/blob/c73232d08f84e110707627a23ceae14d2b534889/src/go/build/read.go#L174-L273 is the code that reads embed comments if we want to track them.

**Comment by ShoshinNikita** [comment]
I think this issue should be closed according to the commit https://github.com/golang/tools/commit/6422c5c8c7ed4d16817458e4be083e2807c874b3:

> Fixes golang/go#38732, golang/go#44342

**Comment by stamblerre** [comment]
Oh good catch, thanks @ShoshinNikita! Closing.
